File: short_problems/C++/find_digits.cpp
#include <iostream>

using namespace std;

int find_digits(long long num) {
    int count = 0;
    long long tmp = num;
    while(tmp) {
        int digit = tmp % 10;
        if (digit) {
            count += num % digit == 0;
        }
        tmp /= 10;
    }
    return count;
}

int main()
{
    int test_cases;
    cin >> test_cases;
    for (int i = 0; i < test_cases; i++) {
        long long num;
        cin >> num;
        cout << find_digits(num) << endl;
    }
    return 0;
}

File: short_problems/Java/FindDigits.java
import java.util.Scanner;

public class FindDigits {

    public static int findDigits(long num) {
        int count = 0;
        long tmp = num;
        while (tmp != 0) {
            long digit = tmp % 10;
            if (digit != 0) {
                count += num % digit == 0 ? 1 : 0;
            }
            tmp /= 10;
        }
        return count;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int numCases = sc.nextInt();
        for (int i = 0; i < numCases; i++) {
            long num = sc.nextLong();
            System.out.println(findDigits(num));
        }
    }
}
